src/backend.rs

```diff
@@ -193,6 +196,32 @@ impl Mock {
 	pub fn set_read_timeout_error(&mut self, err: Option<io::Error>) {
 		self.set_read_timeout_error = err;
 	}
+
+	/// Set the callback function to generate custom replies.
+	///
+	/// The first argument is the message buffer for replies, the second the message itself.
+	///
+	/// # Example
+	///
+	/// ```
+	/// # use zproto::ascii::Port;
+	/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+	/// let mut port = Port::open_mock();
+	/// port.backend_mut().set_reply_callback(|buf, msg|
+	///     buf.extend_from_slice(match msg {
+	///         b"/1 io get ai 1\n" => b"@01 0 OK BUSY -- 5.5\r\n",
+	///         b"/get pos\n" => b"@01 0 OK BUSY -- 20\r\n@02 0 OK BUSY -- 10.1\r\n",
+	///         _ => panic!("unexpected messsage"),
```

Comment:
@marco-swz: CI flagged this spelling mistake ("message" not "messsage"), which blocks this from being merged.